lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -184,13 +210,28 @@ function dosomething_rogue_handle_failure($user, $values, $response = NULL, $e =
         'why_participated' => $values['why_participated'],
         'file' => $values['file'],
         'caption' => $values['caption'],
+        'type' => $values['type'],
         'timestamp' => REQUEST_TIME,
         'response_code' => $response->code, // @TODO: this is currently null until there a better way to get it from Gateway
         'response_values' => (isset($e)) ? $e->getMessage() : NULL,
       ])
       ->execute();
 
-      watchdog('dosomething_rogue', 'reportback not migrated to Rogue', ['user' => $user->uid, 'campaign_id' => $values['nid'], 'campaign_run_id' => $run->nid], WATCHDOG_ERROR);
+    watchdog('dosomething_rogue', 'reportback not migrated to Rogue', ['user' => $user->uid, 'campaign_id' => $values['nid'], 'campaign_run_id' => $run->nid], WATCHDOG_ERROR);
+  } else {
+    db_insert('dosomething_rogue_failed_task_log')
+      ->fields([
+        'rogue_item_id' => $values['rogue_reportback_item_id'],
+        'status' => $values['status'],
+        'type' => $values['type'],
+        'timestamp' => REQUEST_TIME,
+        'response_code' => (isset($response->code)) ? $response->code : NULL, // @TODO: this is currently null until there a better way to get it from Gateway
```

Comment:
🙌  would you mind changing the one above to be like this as well while you're at it?